src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -899,12 +899,12 @@ void TemporalMemory::setPredictedSegmentDecrement(Permanence predictedSegmentDec
   predictedSegmentDecrement_ = predictedSegmentDecrement;
 }
 
-Permanence TemporalMemory::getMaxSegmentsPerCell() const
+UInt TemporalMemory::getMaxSegmentsPerCell() const
```

Comment:
Can we do this as a separate issue? The entire TM and SP use `UInt` (which is just typedef'd to `UInt32` with our current build config). If we want to change them to explicit UInt32, let's do it all together.